extensions/chathistory.md

```diff
@@ -0,0 +1,88 @@
+
+#### Errors and Warnings
+If the server receives an improperly formatted `CHATHISTORY` command, the `CMD_INVALID` error code should be returned.
+
+If the absolute value of `message_count` exceeds the `max_message_count`, warn code `MAX_MSG_COUNT_EXCEEDED` should be returned. The command should continue to be processed as described above.
+
+If no `chathistory` exists to return, the server should return the appropriate error code. `ACCESS_DENIED` should be sent if the user requests content they do not have permission to view.
+
+### Examples
+The examples below are written with `draft/msgid` and `draft/label` tags included. These tags are recommended.
+
+#### Begin
+    @draft/label=ID :irc.host BATCH +ID chathistory target
+#### PRIVMSG
+    @batch=ID;draft/msgid=ID;time=YYYY-MM-DDThh:mm:ss.sssZ :nick!ident@host PRIVMSG target :message
+#### NOTICE
+    @batch=ID;draft/msgid=ID;time=YYYY-MM-DDThh:mm:ss.sssZ :nick!ident@host NOTICE target :message
+#### End
+    :irc.host BATCH -ID
+#### Error
+    @draft/label=ID :nick!ident@host CHATHISTORY ERR ERROR_CODE
+#### Warning
+    @draft/label=ID :nick!ident@host CHATHISTORY WARN WARN_CODE
+
+## Use Cases
+The batch type and supporting command are useful for allowing an "infinite scroll" type capability within the client. A client will, upon scrolling to the top of the active window or a manual trigger, may request `chathistory` from the server and, after receiving returned content, append it to the top of the window. Users can repeat this historic scrolling to retrieve prior history until limitations are met (see below).
+
+## Limitations
+Logging of messages and other actions must be enabled server-side and can be stored in any format desired, given appropriate software exists to retrieve and format such stored contnet. Scrollback can only be retrieved as far back as logs exist for the requesting user in the specified channel or query.
+
+A method for securely identifying the requesting user must exist to ensure content is sent only to the appropriate users and clients. See below for more information.
+
+## Security Considerations
+Secure identification of users and clients must exist in order to ensure that users cannot obtain history that does not belong to them. Use of account names, internal account identifiers, or certificate fingerprints should be strongly considered when matching content to users. The server must verify the current user's identity matches that of the desired content. This information is not sent as part of the `CHATHISTORY` command and must be validated via other means, such as those stated above. Access must be checked first and return an `ACCESS_DENIED` error as appropriate. If no `ACCESS_DENIED` error exists, the server can check for returnable content.
```

Comment:
> This limitation exists to prevent retrieval of content you weren't in the channel for.

Shouldn't this be a consideration on an IRC server and not rules enforced by the specification? An implementation of `CHATHISTORY` might want to provide public history for some channels. For example some private IRC server at a company where all employees could join later and view existing history.